src/RevSysCalls.cc

```diff
@@ -486,25 +491,31 @@ EcallStatus RevProc::ECALL_openat(RevInst& inst){
    */
 
   /* Read the filename from memory one character at a time until we find '\0' */
+  auto& Thread = GetThreadOnHart(HartToExec);
+
+  auto& EcallState = Harts.at(HartToExec)->GetEcallState();
 
   auto action = [&]{
     // Do the openat on the host
-    dirfd = open(std::filesystem::current_path().c_str(), O_RDONLY);
-    int fd = openat(dirfd, Harts.at(HartToExec)->GetEcallState().string.c_str(), O_RDWR);
+    dirfd = open(std::filesystem::current_path().c_str(), O_RDWR);
+    int fd = openat(dirfd, EcallState.string.c_str(), O_RDWR);
 
```

Comment:
Do we ignore any other `openat` syscall parameters and always open a file RDWR?

Reply:
That's how I did it originally way back when but I suppose there's no reason to keep it that way